allennlp/modules/token_embedders/embedding.py

```diff
@@ -180,9 +188,33 @@ def extend_vocab(self,  # pylint: disable=arguments-differ
             vocab_namespace = "tokens"
             logging.warning("No vocab_namespace provided to Embedder.extend_vocab. Defaulting to 'tokens'.")
 
+        extended_num_embeddings = extended_vocab.get_vocab_size(vocab_namespace)
+        if extended_num_embeddings <= self.num_embeddings:
+            # It's already been extended. No need to initialize / read pretrained file in first place (no-op)
+            return
+
+        if self._pretrained_file and os.path.exists(self._pretrained_file):
+            # pretrained_file attribute was saved during training and is available:
+            if pretrained_file:
+                # But user also passed pretrained_file
+                model_path_info = "at model_path, {model_path} " if model_path else ""
+                logging.warning(f"You passed pretrained_file, {pretrained_file} for "
+                                f"embedding {model_path_info}. But it's ignored because "
+                                f"original pretrained_file, {self._pretrained_file} is available.")
+            pretrained_file = self._pretrained_file
+
+        if not (pretrained_file and os.path.exists(pretrained_file)):
+            extra_info = (f"Originally pretrained_file was at "
+                          f"{self._pretrained_file}. " if self._pretrained_file else "")
+            # It's better to warn here and not give error because there is no way to distinguish between
+            # whether pretrained-file wasn't used during training or user forgot to pass / passed incorrect
+            # mapping. Giving error would prevent former case to executable.
```

Comment:
"Raising an error would prevent fine-tuning in the former case"?

Reply:
We can only raise error if pretrained file wasn't available when we know *for sure* that it was used at time of training. If pre-trained wasn't used in first place, the extension should ideally happen with random vectors and we don't need the file.

For models after this PR, we can just check `_pretrained_file` attribute to know whether pretrained file was used or not and decide whether it's essential or not. However, earlier we didn't store the `_pretrained_file` attribute, so effectively there's no way of really knowing whether pretrained file was used or not. 

If we raise error, user won't be able to extend if pre-trained file is in fact not required.

Reply:
Oops, I think you just wanted me to change the comment to `Raising an error would prevent fine-tuning in the former case`, right? I thought, you want me to explain case!

Reply:
done. 👍